actionpack/lib/action_dispatch/middleware/static.rb

```diff
@@ -48,7 +48,10 @@ def call(env)
         env['PATH_INFO']            = gzip_path
         status, headers, body       = @file_server.call(env)
         headers['Content-Encoding'] = 'gzip'
-        headers['Content-Type']     = content_type(path)
+        if status != 304
+          # 304 response should not include Content-Type header.
```

Comment:
I dont think the comment is necessary here, as the `if` already states that statement .